aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/DownloadCallable.java

```diff
@@ -174,6 +174,8 @@ private void downloadInParallel(int partCount) throws Exception {
             lastFullyMergedPartNumber = 0;
         }
 
+        ServiceUtils.createParentDirectoryIfNecessary(dstfile);
```

Comment:
Why not move it all the way up to line 128? We really don't want to run into these inconsistencies again ;-)